packager/react-packager/src/DependencyResolver/fastfs.js

```diff
@@ -293,16 +293,16 @@ class File {
   }
 
   getFiles() {
-    const files = [];
-    Object.keys(this.children).forEach(key => {
+    const files = Object.keys(this.children).map(key => {
       const file = this.children[key];
       if (file.isDir) {
-        files.push(...file.getFiles());
-      } else {
-        files.push(file);
+        return file.getFiles();
```

Comment:
@cpojer philikon's approach is simple to read and should be close to optimal. Calling `concat` many times on arrays with > 100k elements is going to invoke a lot of extra copying unless V8 is using persistent data structures or CoW under the hood.